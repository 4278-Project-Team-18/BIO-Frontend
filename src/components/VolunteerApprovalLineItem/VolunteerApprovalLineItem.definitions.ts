import type { Volunteer } from "../../interfaces/Users.interface";

export interface VolunteerApprovalLineItemProps {
  volunteer: Volunteer;
}
