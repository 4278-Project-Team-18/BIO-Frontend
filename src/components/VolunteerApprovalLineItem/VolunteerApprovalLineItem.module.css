@media (min-width: 600px) {
  .line-item-container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    padding: 5px;
  }

  .line-item-container-left {
    display: flex;
    justify-content: left;
    align-items: center;
    padding: 5px;
  }

  .line-item-name-icon {
    font-size: 24px;
    padding-left: 10px;
    padding-right: 30px;
  }

  .line-item-name {
    font-weight: 400;
    font-size: 16px;
    color: #565656;
    justify-content: left;
  }

  .line-item-container-right {
    display: flex;
    justify-content: left;
    align-items: center;
    padding: 5px;
  }

  .line-item-approve-button {
    display: flex;
    justify-content: left;
    align-items: center;
    padding-top: 2px;
    margin-left: 12px;
    background: #bdffc8;
  }

  .line-item-deny-button {
    display: flex;
    justify-content: left;
    align-items: center;
    padding-top: 2px;
    margin-left: 12px;
    background-color: #ffb4b4;
  }

  .line-item-button-title {
    font-weight: 300;
    font-size: 14px;
    color: #434343;
    font-family: "Poppins", sans-serif;
    font-weight: 300;
  }

  .line-item-button-icon {
    font-size: 16px;
    padding-left: 10px;
    padding-bottom: 2px;
    color: #434343;
  }
}

@media (max-width: 599px) {
  .line-item-container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    padding: 5px;
  }

  .line-item-container-left {
    display: flex;
    justify-content: left;
    align-items: center;
    padding: 5px;
  }

  .line-item-name-icon {
    font-size: 24px;
    padding-left: 10px;
    padding-right: 30px;
  }

  .line-item-name {
    font-weight: 400;
    font-size: 14px;
    color: #565656;
    justify-content: left;
  }

  .line-item-container-right {
    display: flex;
    justify-content: left;
    align-items: center;
    padding: 5px;
  }

  .line-item-approve-button {
    display: flex;
    justify-content: left;
    align-items: center;
    padding-top: 2px;
    margin-left: 12px;
    background: #bdffc8;
  }

  .line-item-deny-button {
    display: flex;
    justify-content: left;
    align-items: center;
    padding-top: 2px;
    margin-left: 12px;
    background-color: #ffb4b4;
  }

  .line-item-button-title {
    font-weight: 300;
    font-size: 14px;
    color: #434343;
    font-family: "Poppins", sans-serif;
    font-weight: 300;
  }

  .line-item-button-icon {
    font-size: 16px;
    padding-left: 10px;
    padding-bottom: 2px;
    color: #434343;
  }
}
