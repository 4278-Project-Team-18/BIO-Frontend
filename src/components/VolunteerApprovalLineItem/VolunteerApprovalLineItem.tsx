import styles from "./VolunteerApprovalLineItem.module.css";
import VolunteerUpdateApprovalStatus from "../VolunteerUpdateApprovalStatus/VolunteerUpdateApprovalStatus";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faClock } from "@fortawesome/free-regular-svg-icons";
import type { VolunteerApprovalLineItemProps } from "./VolunteerApprovalLineItem.definitions";

const VolunteerApprovalLineItem = ({
  volunteer,
}: VolunteerApprovalLineItemProps) => (
  <div className={styles["line-item-container"]}>
    <div className={styles["line-item-container-left"]}>
      <FontAwesomeIcon
        icon={faClock}
        className={styles["line-item-name-icon"]}
      />
      <div className={styles["line-item-name"]}>
        {`${volunteer.firstName} ${volunteer.lastName}`}
      </div>
    </div>
    <div className={styles["line-item-container-right"]}>
      <VolunteerUpdateApprovalStatus volunteer={volunteer} />
    </div>
  </div>
);

export default VolunteerApprovalLineItem;
