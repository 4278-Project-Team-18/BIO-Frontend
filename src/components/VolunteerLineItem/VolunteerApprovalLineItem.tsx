import styles from "./VolunteerApprovalLineItem.module.css";
import { faCheck, faClose } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import type { VolunteerApprovalLineItemProps } from "./VolunteerApprovalLineItem.definitions";

const VolunteerApprovalLineItem = ({
  volunteer,
}: VolunteerApprovalLineItemProps) => (
  <div className={styles["line-item-container"]}>
    <div className={styles["line-item-container-left"]} />
    <div className={styles["line-item-name"]}>
      {`${volunteer.firstName} ${volunteer.lastName}`}
    </div>
    <div className={styles["line-item-container-right"]}>
      <button className={styles["line-item-approve-button"]}>
        <div className={styles["line-item-button-title"]}>Approve</div>
        <FontAwesomeIcon icon={faCheck} className={styles["line-item-icon"]} />
      </button>
      <button className={styles["line-item-deny-button"]}>
        <div className={styles["line-item-button-title"]}>Deny</div>
        <FontAwesomeIcon icon={faClose} className={styles["line-item-icon"]} />
      </button>
    </div>
  </div>
);

export default VolunteerApprovalLineItem;
