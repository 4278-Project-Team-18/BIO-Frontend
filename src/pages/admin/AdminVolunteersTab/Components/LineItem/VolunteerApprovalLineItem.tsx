import styles from "./VolunteerApprovalLineItem.module.css";

interface VolunteerApprovalLineItemProps {
  volunteer: string;
}

const VolunteerApprovalLineItem = ({
  volunteer,
}: VolunteerApprovalLineItemProps) => (
  <div className={styles["container"]}>
    <div className={styles["volunteer"]}>{volunteer}</div>
    <div className={styles["buttons"]}>
      <button>Approve</button>
      <button>Deny</button>
    </div>
  </div>
);

export default VolunteerApprovalLineItem;
