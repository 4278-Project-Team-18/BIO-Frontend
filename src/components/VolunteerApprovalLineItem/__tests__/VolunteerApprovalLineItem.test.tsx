import VolunteerApprovalLineItem from "../VolunteerApprovalLineItem";
import { render } from "@testing-library/react";
import type { Volunteer } from "../../../interfaces/User.interface";

// mock clerk use auth and getToken to return null
jest.mock("@clerk/clerk-react", () => ({
  useAuth: () => ({ getToken: () => "" }),
}));

describe("VolunteerApprovalLineItem Tests", () => {
  test("VolunteerApprovalLineItem should render correctly", () => {
    const mockVolunteer = {
      _id: "12345",
      firstName: "Test",
      lastName: "Test",
      email: "testemail@gmail.com",
    } as Volunteer;

    const component = render(
      <VolunteerApprovalLineItem volunteer={mockVolunteer} />,
    );
    expect(component).toBeTruthy();

    // Check if the name is rendered
    expect(component.getByText("Test Test")).toBeTruthy();

    // Check if the buttons are rendered
    expect(component.getByText("Approve")).toBeTruthy();
    expect(component.getByText("Deny")).toBeTruthy();
  });

  test("VolunteerApprovalLineItem snapshot should render correctly", () => {
    const mockVolunteer = {
      _id: "12345",
      firstName: "Test",
      lastName: "Test",
      email: "testemail@gmail.com",
    } as Volunteer;

    const component = render(
      <VolunteerApprovalLineItem volunteer={mockVolunteer} />,
    );
    expect(component).toMatchSnapshot();
  });
});
