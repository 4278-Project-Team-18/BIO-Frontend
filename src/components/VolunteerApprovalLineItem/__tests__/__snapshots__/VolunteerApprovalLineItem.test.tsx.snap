// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VolunteerApprovalLineItem Tests VolunteerApprovalLineItem snapshot should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="line-item-container"
      >
        <div
          class="line-item-container-left"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-clock line-item-name-icon"
            data-icon="clock"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 256A208 208 0 1 1 48 256a208 208 0 1 1 416 0zM0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM232 120V256c0 8 4 15.5 10.7 20l96 64c11 7.4 25.9 4.4 33.3-6.7s4.4-25.9-6.7-33.3L280 243.2V120c0-13.3-10.7-24-24-24s-24 10.7-24 24z"
              fill="currentColor"
            />
          </svg>
          <div
            class="line-item-name"
          >
            Test Test
          </div>
        </div>
        <div
          class="line-item-container-right"
        >
          <div
            class="buttons-container"
          >
            <button
              class="loading-button"
              style="margin-left: 1rem; background-color: rgb(90, 208, 109);"
              type="button"
            >
              <span
                class="loading-button-label"
              >
                Approve
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-check loading-button-spinner"
                data-icon="check"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              class="loading-button"
              style="margin-left: 1rem; background-color: rgb(239, 68, 68);"
              type="button"
            >
              <span
                class="loading-button-label"
              >
                Deny
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-xmark loading-button-spinner"
                data-icon="xmark"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="line-item-container"
    >
      <div
        class="line-item-container-left"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-clock line-item-name-icon"
          data-icon="clock"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 256A208 208 0 1 1 48 256a208 208 0 1 1 416 0zM0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM232 120V256c0 8 4 15.5 10.7 20l96 64c11 7.4 25.9 4.4 33.3-6.7s4.4-25.9-6.7-33.3L280 243.2V120c0-13.3-10.7-24-24-24s-24 10.7-24 24z"
            fill="currentColor"
          />
        </svg>
        <div
          class="line-item-name"
        >
          Test Test
        </div>
      </div>
      <div
        class="line-item-container-right"
      >
        <div
          class="buttons-container"
        >
          <button
            class="loading-button"
            style="margin-left: 1rem; background-color: rgb(90, 208, 109);"
            type="button"
          >
            <span
              class="loading-button-label"
            >
              Approve
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-check loading-button-spinner"
              data-icon="check"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
                fill="currentColor"
              />
            </svg>
          </button>
          <button
            class="loading-button"
            style="margin-left: 1rem; background-color: rgb(239, 68, 68);"
            type="button"
          >
            <span
              class="loading-button-label"
            >
              Deny
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark loading-button-spinner"
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
